Sming/SmingCore/Network/TelnetServer.cpp

```diff
@@ -0,0 +1,139 @@
+/*
+ * telnetServer.cpp
+ *
+ *  Created on: 18 apr. 2015
+ *      Author: Herman
+ */
+
+#include "TelnetServer.h"
+#include "TcpServer.h"
+
+TelnetServer::TelnetServer() : TcpServer()
+{
+	// TODO Auto-generated constructor stub
+
+}
+
+TelnetServer::~TelnetServer()
+{
+	// TODO Auto-generated destructor stub
+}
+
+void TelnetServer::onClient(TcpClient *client)
+{
+	debugf("TelnetServer onClient %s", client->getRemoteIp().toString().c_str() );
+	curClient = client;
+	TcpServer::onClient(client);
+	curClient->sendString("Welcome to Sming / ESP6266 Telnet\r\n");
+}
+
+void TelnetServer::onClientComplete(TcpClient& client, bool succesfull)
+{
+	curClient = nullptr;
+	debugf("TelnetServer onClientComplete %s", client.getRemoteIp().toString().c_str() );
+	TcpServer::onClientComplete(client, succesfull);
+}
+
+void TelnetServer::wrchar(char c)
+{
+	char ca[2];
+	ca[0]= c;
+	curClient->write(ca,1);
+}
+
+bool TelnetServer::onClientReceive (TcpClient& client, char *data, int size)
+{
+	debugf("TelnetServer onClientReceive : %s, %d bytes \r\n",client.getRemoteIp().toString().c_str(),size );
+	debugf("Data : %s", data);
+
+	for (int i=0;i<size;i++)
+	{
+		if (data[i] >= 240) // part of Telnet command
+		{
+			debugf("TelnetServer : Telnet Command ignored\r\n");
+		}
+		else
+		{
+			if (commandIndex < TELNETSERVER_MAX_COMMANDSIZE)
+			{
+				commandBuf[commandIndex++] = data[i];
+				if ( data[i] == '\n')
+				{
+					commandBuf[commandIndex] = 0;
+					processCommandLine (commandBuf, commandIndex -1);
+					commandIndex = 0;
+				}
+			}
+		}
+	}
+	curClient->send(prompt,1);
+	return true;
+}
+
+void TelnetServer::processCommandLine( char *cmdBuf, int size)
+{
+	debugf("Telnet Command %s", cmdBuf);
```

Comment:
Yes, sure. I just included some very rudimentary commands to show some functionality.
definitely not as the final implementation. My focus in this PR was on getting TelnetSever working on top of/together with TcpServer and TcpClient